### ISSUE TITLE
Using output as a separate acitvation layer causing very poor accuracy.

### ISSUE THREAD

**Issue by Ehsan1997** [keras-team/keras#13621]
Following are the last two lines of my model:
```
outputs = Dense(4)(act_fc1)
predictions = Activation(activation='softmax')(outputs)
```

If I use predictions as the model output, the model does not learn anything, but if I remove the predictions layer and add the activations parameter to outputs Dense layer, it works fine.

**Comment by amarlearning** [comment]
@Ehsan1997 `predictions` here is just an object of Activation class. It's not the part of the model.

I would suggest you  to build model like this:
```
model = Sequential()
model.add(Dense(512, activation='relu')
model.add(Dropout(0.5))
model.add(Dense(256, activation='relu'))
model.add(Dropout(0.25))
model.add(Dense(10, activation='softmax'))
```
In this way you not get confused. You can also print your visualize you model using this code for cross-check.
```
from keras.utils import plot_model
plot_model(model, to_file='model.png')
```

**Comment by Ehsan1997** [comment]
> @Ehsan1997 `predictions` here is just an object of Activation class.
> 
> I would suggest you to build model like this:
> 
> ```
> model = Sequential()
> model.add(Dense(512, activation='relu')
> model.add(Dropout(0.5))
> model.add(Dense(256, activation='relu'))
> model.add(Dropout(0.25))
> model.add(Dense(10, activation='softmax'))
> ```
> 
> In this way you will not get confused. You can also visualize your model using this code to cross-check.
> 
> ```
> from keras.utils import plot_model
> plot_model(model, to_file='model.png')
> ```

Could you please explain why isn't it part of the network??

**Comment by Ehsan1997** [comment]
Okay it seems as if there was some problem with colab's version of keras, I just changed it to tensorflow.keras and now everything seems fine.
